### PR TITLE
Custom fetch hook

### PR DESCRIPTION
**Pull request by keepforever**
Thought these hooks might be useful additions to `Gateway`. 

The idea would be to use what's there in `useFetchById` as boiler and adjust as needed.

### CODE REVIEW

File: src/hooks/useFetchById.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    const [isError, setIsError] = useState(false);
+
+    useEffect(() => {
+        async function fetchData(id) {
+            let res = null;
+            setIsLoading(true);
+            try {
+                res = await axios.request({
+                    method: 'GET',
+                    url: `https://api.findawayworld.com/v4/audiobooks/${id}`,
```
Comment (cwardzala) [inline src/hooks/useFetchById.js]: this does not allow the api to be changed based on the project/use-case. I would adopt a different API, something that accepts the url and a headers object, maybe query-params as well.

```js
useAxiosGet('/my/url', headers, query_params)
```

File: src/hooks/useFetchById.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    return [data, isLoading, isError];
+};
+
+export default useFetchById;
+
+/*
+    Demo use
+*/
+export const MyContrivedExample = props => {
```
Comment (cwardzala) [inline src/hooks/useFetchById.js]: I would move this to the utils page, as a code block, or comment it out here as an example. Regardless please update the utils page with this.

File: src/hooks/useWhyDidYouUpdate.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        // Finally update previousProps with current props for next hook call
+        previousProps.current = props;
+    });
+}
+
+/*
+    To implement this helper hook:
+*/
+const MyFunctionalComponent = props => {
```
Comment (cwardzala) [inline src/hooks/useWhyDidYouUpdate.js]: Add this to the utils page as a code block example or comment out here as an example.

File: src/hooks/useWhyDidYouUpdate.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    const previousProps = useRef();
+
+    useEffect(() => {
+        if (previousProps.current) {
+            // Get all keys from previous and current props
+            const allKeys = Object.keys({ ...previousProps.current, ...props });
+            // Use this object to keep track of changed props
+            const changesObj = {};
+            // Iterate through keys
+            allKeys.forEach(key => {
```
Comment (cwardzala) [inline src/hooks/useWhyDidYouUpdate.js]: You could probably use `reduce` here to cleanup the logic.

```js
const changesObj = allKeys.reduce((obj, key) => {
    // If previous is different from current
    if (previousProps.current[key] !== props[key]) {
        // Add to changesObj
        obj[key] = {
            from: previousProps.current[key],
            to: props[key],
        };
    }
    return obj;
}, {});
```

**Comment by keepforever** [comment]
Closing pull request because proposed addition does not fit vison of repo

**Pull request closed without merging**